modules/nlu/src/backend/api.ts

```diff
@@ -51,27 +50,36 @@ export default async (bp: typeof sdk, state: NLUState) => {
       return res.status(404).send(`Bot ${botId} doesn't exist`)
     }
 
-    const modelId = botNLU.modelsByLang[botNLU.defaultLanguage]
+    const predictionLanguage = botNLU.defaultLanguage
+    const modelId = botNLU.modelsByLang[predictionLanguage]
 
     try {
       // TODO: language should be a path param of route
 
-      let nlu: sdk.IO.EventUnderstanding
+      let nlu: sdk.NLU.PredictOutput
 
       const spellChecked = await state.engine.spellCheck(value.text, modelId)
+
+      const t0 = Date.now()
       if (spellChecked !== value.text) {
-        const originalPrediction = (await state.engine.predict(value.text, value.contexts, modelId)) as PredictOutput
-        const spellCheckedPrediction = (await state.engine.predict(
-          spellChecked,
-          value.contexts,
-          modelId
-        )) as PredictOutput
+        const originalPrediction = await state.engine.predict(value.text, modelId)
+        const spellCheckedPrediction = await state.engine.predict(spellChecked, modelId)
         nlu = mergeSpellChecked(originalPrediction, spellCheckedPrediction)
       } else {
-        nlu = await state.engine.predict(value.text, value.contexts, modelId)
+        nlu = await state.engine.predict(value.text, modelId)
+      }
+      const ms = Date.now() - t0
+
+      const event: sdk.IO.EventUnderstanding = {
+        ...nlu,
+        includedContexts: value.contexts,
+        language: predictionLanguage,
+        detectedLanguage: undefined,
```

Comment:
detectedLanguage always undefined? if it's always undefined does that mean that we don't use it anywhere ?  not even in the nlu summary in the UI ? 

if yes then why not remove it all along from eventUnderstanding ?

Reply:
Well, this is only the `/predict` route, used by nlu-testing...  In the nlu middleware, the language detection is ran and the field is not empty

Reply:
I see